core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -210,6 +211,22 @@ private TableProperties() {
   public static final String SPARK_WRITE_PARTITIONED_FANOUT_ENABLED = "write.spark.fanout.enabled";
   public static final boolean SPARK_WRITE_PARTITIONED_FANOUT_ENABLED_DEFAULT = false;
 
+  public static final String PARTITIONED_FANOUT_WRITERS_CACHE_SIZE = "write.partition.fanout.writers-cache-size";
+  public static final int PARTITIONED_FANOUT_WRITERS_CACHE_SIZE_DEFAULT = Integer.MAX_VALUE;
+
+  public static final String PARTITIONED_FANOUT_WRITERS_CACHE_EVICT_MS =
+      "write.partition.fanout.writers-cache-eviction-ms";
+  public static final long PARTITIONED_FANOUT_WRITERS_CACHE_EVICT_MS_DEFAULT =
+      TimeUnit.MINUTES.toMillis(Long.MAX_VALUE);
+
+  public static final String PARTITIONED_DELTA_WRITERS_CACHE_SIZE = "write.partition.delta.writers-cache-size";
+  public static final int PARTITIONED_DELTA_WRITERS_CACHE_SIZE_DEFAULT = Integer.MAX_VALUE;
+
+  public static final String PARTITIONED_DELTA_WRITERS_CACHE_EVICT_MS =
+      "write.partition.delta.writers-cache-eviction-ms";
+  public static final long PARTITIONED_DELTA_WRITERS_CACHE_EVICT_MS_DEFAULT =
+      TimeUnit.MINUTES.toMillis(Long.MAX_VALUE);
```

Comment:
I think these options are specific to the runtime environment and should not be table properties.